staging/src/k8s.io/client-go/tools/lease/controller.go

```diff
@@ -210,26 +221,12 @@ func (c *controller) newLease(base *coordinationv1.Lease) *coordinationv1.Lease
 	}
 	lease.Spec.RenewTime = &metav1.MicroTime{Time: c.clock.Now()}
 
-	// Setting owner reference needs node's UID. Note that it is different from
-	// kubelet.nodeRef.UID. When lease is initially created, it is possible that
-	// the connection between master and node is not ready yet. So try to set
-	// owner reference every time when renewing the lease, until successful.
-	if len(lease.OwnerReferences) == 0 {
-		if node, err := c.client.CoreV1().Nodes().Get(context.TODO(), c.holderIdentity, metav1.GetOptions{}); err == nil {
-			lease.OwnerReferences = []metav1.OwnerReference{
-				{
-					APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
-					Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
-					Name:       c.holderIdentity,
-					UID:        node.UID,
-				},
-			}
-		} else {
-			klog.Errorf("failed to get node %q when trying to set owner ref to the node lease: %v", c.holderIdentity, err)
-		}
+	if c.newLeasePostProcessFunc != nil {
```

Comment:
As discussed on the other PR - please revert to the existing behavior of setting owner ref here.

Reply:
Never mind - I see that you made that more generic and it's here.